checker/src/main/java/org/checkerframework/checker/resourceleak/MustCallInferenceLogic.java

```diff
@@ -43,88 +76,620 @@ public class MustCallInferenceLogic {
+      }
+    }
+  }
+
+  /**
+   * Updates a set of obligations based on an assignment statement node. If the left-hand side of
+   * the assignment is a field that is an "enclosed owning field", adds the owning field to the
+   * method's parameters if its alias is assigned to the field. If the left-hand side of the
+   * assignment is a resource variable and the right-hand side is a must-call-close method call,
+   * adds the owning resource to the method's parameters if its alias is assigned to the resource
+   * variable. Otherwise, updates the obligations based on the assignment.
+   *
+   * @param obligations The set of obligations to update.
+   * @param assignmentNode The assignment statement node.
+   */
+  private void updateObligationsForAssignment(
+      Set<Obligation> obligations, AssignmentNode assignmentNode) {
+    Node lhs = assignmentNode.getTarget();
+    Element lhsElement = TreeUtils.elementFromTree(lhs.getTree());
+    // Use the temporary variable for the rhs if it exists.
+    Node rhs = NodeUtils.removeCasts(assignmentNode.getExpression());
+    rhs = mcca.getTempVarOrNode(rhs);
+
+    if (!(rhs instanceof LocalVariableNode)) {
+      return;
+    }
+    Obligation rhsObligation =
+        MustCallConsistencyAnalyzer.getObligationForVar(obligations, (LocalVariableNode) rhs);
+    if (rhsObligation == null) {
+      return;
+    }
+
+    if (lhsElement.getKind() == ElementKind.FIELD) {
+      if (!getEnclosedOwningFields().contains(lhsElement)) {
+        return;
+      }
+
+      if (TreeUtils.isConstructor(enMethodTree)) {
+        addOwningToParamsIfDisposedAtAssignment(obligations, rhsObligation, rhs);
+      } else {
+        if (owningFieldToECM.contains((VariableElement) lhsElement)) {
+          owningFieldToECM.remove((VariableElement) lhsElement);
+        }
+        addOwningToParamsIfDisposedAtAssignment(obligations, rhsObligation, rhs);
+      }
+
+    } else if (lhsElement.getKind() == ElementKind.RESOURCE_VARIABLE && mcca.isMustCallClose(rhs)) {
+      addOwningToParamsIfDisposedAtAssignment(obligations, rhsObligation, rhs);
+    } else if (lhs instanceof LocalVariableNode) {
+      LocalVariableNode lhsVar = (LocalVariableNode) lhs;
+      Obligation lhsObligation =
+          MustCallConsistencyAnalyzer.getObligationForVar(obligations, lhsVar);
+      if (lhsObligation == null) {
+        Set<ResourceAlias> newResourceAliasesForObligation =
+            new LinkedHashSet<>(rhsObligation.resourceAliases);
+        newResourceAliasesForObligation.add(
+            new ResourceAlias(new LocalVariable(lhsVar), lhs.getTree()));
+        obligations.remove(rhsObligation);
+        obligations.add(new Obligation(newResourceAliasesForObligation));
+      } else {
+        obligations.remove(rhsObligation);
+      }
+    }
+  }
+
+  /**
+   * Adds owning fields to the method's parameters, if its alias referred to by the right-hand side
+   * of the assignment is disposed of during the assignment.
```

Comment:
What does "disposed of" mean here? Is that a synonym for the obligation being fulfilled?